VmAgent.Core.UnitTets/SystemOperationsTests.cs

```diff
@@ -0,0 +1,293 @@
+        {
+            return toCompare.FullName == path;
+        }
+
+        [TestMethod]
+        [TestCategory("BVT")]
+        public void TestSetUnixOwnerRecursive()
+        {
+            List<string> expectedItemsToOwn = new List<string>()
+            {
+                _directoryPath,
+                _subdirectoryPath,
+                _directoryFilePath,
+                _subdirectoryFilePath
+            };
+
+            _systemOperations.SetUnixOwnerIfNeeded(_directoryPath, true);
+
+            expectedItemsToOwn.ForEach(
+                    (item) => _mockFileSystemOperations.Verify(x => x.SetUnixOwner(It.Is<string>((path) => path == item), _systemOperations.User), Times.Once));
+        }
+
+        [TestMethod]
+        [TestCategory("BVT")]
+        public void TestSetUnixOwnerNonRecursive()
+        {
+            List<string> unExpectedItemsToOwn = new List<string>()
+            {
+                _subdirectoryPath,
+                _directoryFilePath,
+                _subdirectoryFilePath
+            };
+
+            _systemOperations.SetUnixOwnerIfNeeded(_directoryPath, false);
+
+            unExpectedItemsToOwn.ForEach(
+                    (item) => _mockFileSystemOperations.Verify(x => x.SetUnixOwner(It.Is<string>((path) => path == item), _systemOperations.User), Times.Never));
+            _mockFileSystemOperations.Verify(x => x.SetUnixOwner(It.Is<string>((path) => path.EndsWith(_directoryPath)), _systemOperations.User), Times.Once);
+        }
+
+        [TestMethod]
+        [TestCategory("BVT")]
+        public void TestSetUnixOwnerDoesNothing()
+        {
+            List<string> expectedItemsToOwn = new List<string>()
+            {
+                _directoryPath,
+                _subdirectoryPath,
+                _directoryFilePath,
+                _subdirectoryFilePath
+            };
+            VmConfiguration config = new VmConfiguration(56001, "vmid", new VmDirectories("root"), false);
+            SystemOperations systemOperations = new SystemOperations(config, _logger, _mockFileSystemOperations.Object);
+            systemOperations.SetUnixOwnerIfNeeded(_directoryPath, true);
+
+            expectedItemsToOwn.ForEach(
+                    (item) => _mockFileSystemOperations.Verify(x => x.SetUnixOwner(It.IsAny<string>(), It.IsAny<string>()), Times.Never));
+        }
+
+        [TestMethod]
+        [TestCategory("BVT")]
+        public void TestCreateParentStructure()
+        {
+            List<string> expectedDirectoriesToCreateOwn = new List<string>()
+            { 
+                _subdirectoryPath,
+                _directoryPath
+            };
+
+            _systemOperations.CreateDirectoryAndParents(_subDirectoryInfo);
+            expectedDirectoriesToCreateOwn.ForEach(
+                    (item) => { 
+                        _mockFileSystemOperations.Verify(x => x.SetUnixOwner(It.Is<string>((path) => path == item), _systemOperations.User), Times.Once);
```

Comment:
Having a helper method to verify this action with the Times passed in will limit a lot of code duplication.

Reply:
I am partial to having the Assertions of the method clearly inside the method.